src/services/services.ts

```diff
@@ -2478,6 +2482,51 @@ export function createLanguageService(
         }
     }
 
+    function getLinkedEditingAtPosition(fileName: string, position: number): LinkedEditingInfo | undefined {
+        const sourceFile = syntaxTreeCache.getCurrentSourceFile(fileName);
+        const token = findPrecedingToken(position, sourceFile);
+        if (!token) return undefined;
+
+        if (isJsxFragment(token.parent.parent)) {
+            const openPos = token.parent.parent.openingFragment.getStart(sourceFile) + 1; // "<".length
+            const closePos = token.parent.parent.closingFragment.getStart(sourceFile) + 2; // "</".length
+
+            // only allows mirroring right after opening bracket: <| ></| >
+            if ((position !== openPos) && (position !== closePos)) return undefined;
+
+            const ranges = [{ start: openPos, length: 0 }, { start: closePos, length: 0 }];
+            const wordPattern = undefined;
+            return { ranges, wordPattern };
+        }
+        else {
+            // looks for mirroring in element tag names
+            const tag = findAncestor(token,
+                n => {
+                    if (!n.parent.parent) return "quit";
+                    else if (isJsxElement(n.parent.parent)) {
+                        if (isJsxOpeningElement(n.parent) || isJsxClosingElement(n.parent)) {
+                            return true;
+                        }
+                        return "quit";
+                    }
+                    return false;
+                });
+            if (!tag) return undefined;
+
+            const element = tag.parent as JsxOpeningElement | JsxClosingElement;
+            const openTag = { start: element.parent.openingElement.tagName.getStart(sourceFile), end: element.parent.openingElement.tagName.end };
+            const endTag = { start: element.parent.closingElement.tagName.getStart(sourceFile), end: element.parent.closingElement.tagName.end };

```

Comment:
`openTag` and `endTag` are never used except just to "namespace" some variables. I would just create `openTagStart`, `openTagEnd`, `closingTagStart`, and `closingTagEnd`.